index.js

```diff
@@ -499,20 +499,16 @@ function signAsync (opts) {
     })
     .then(function () {
       // Pre-sign operations
-      var promise = Promise.resolve()
-      if (opts.version ? compareVersion(opts.version, '1.1.1') >= 0 : true) {
+      if (opts.entitlements != null && (opts.version == null || compareVersion(opts.version, '1.1.1') >= 0)) {
         // Enable Mac App Store sandboxing without using temporary-exception, introduced in Electron v1.1.1. Relates to electron#5601
         if (opts['pre-auto-entitlements'] === false) {
           debugwarn('Pre-sign operation disabled for entitlements automation.')
         } else {
           debuglog('Pre-sign operation enabled for entitlements automation with versions >= `1.1.1`; disable by setting `pre-auto-entitlements` to `false`.')
-          promise = promise.then(function () {
-            debuglog('Automating entitlement app group...')
-            return preAutoEntitlementAppGroupAsync(opts)
-          })
+          debuglog('Automating entitlement app group...')
+          return preAutoEntitlementAppGroupAsync(opts)
         }
       }
-      return promise
```

Comment:
@develar I don't know but a similar case has caused a slight issue previously... not sure if being a same cause: https://github.com/electron-userland/electron-osx-sign/pull/44/commits/184985a6fa556145d48ab770d4d9c1f36cf758fc.

Probably that's a different case because of returning a promise obj so what following uses it before continuing `.then()`?